build/ncollide2d/tests/geometry/ray_cast.rs

```diff
@@ -88,3 +88,54 @@ fn perpendicular_raycast_starting_bellow_segment() {
     let ray = Ray::new(Point2::new(0.0, -11.0), Vector2::new(1.0, 0.0));
     assert!(!segment.intersects_ray(&na::one(), &ray));
 }
+
+///    Ray Target
+///    +
+/// 3  |     Ray moved up each iteration
+/// |  v     ^
+/// 2  *--+  |
+/// |  |  |  +
+/// 1  +--+  *<----+Ray origin
+/// |
+/// 0--1--2--3
+///
+/// Tests the accuracy of raycaster collision detection against a `ConvexPolygon`.
+#[test]
+fn convexpoly_raycast_fuzz() {
+    let vertices: Vec<Point2<f64>> = vec![[1, 1], [1, 2], [2, 2], [2, 1]]
```

Comment:
This should be `[1, 1], [2, 1], [2, 2], [1, 2]`. This is not mentioned on the doc (I'll fix that) but the vertices must be provided in counterclockwise order.